apps/login/app/(login)/register/idp/[provider]/success/page.tsx

```diff
@@ -0,0 +1,133 @@
+import { ProviderSlug } from "#/lib/demos";
+import { addHumanUser, server } from "#/lib/zitadel";
+import Alert, { AlertType } from "#/ui/Alert";
+import {
+  AddHumanUserRequest,
+  IDPInformation,
+  RetrieveIdentityProviderInformationResponse,
+  user,
+  IDPLink,
+} from "@zitadel/server";
+
+const PROVIDER_MAPPING: {
+  [provider: string]: (rI: IDPInformation) => Partial<AddHumanUserRequest>;
+} = {
+  [ProviderSlug.GOOGLE]: (idp: IDPInformation) => {
+    const idpLink: IDPLink = {
+      idpId: idp.idpId,
+      userId: idp.userId,
+      userName: idp.userName,
+    };
+    const req: Partial<AddHumanUserRequest> = {
+      username: idp.userName,
+      email: {
+        email: idp.rawInformation?.User?.email,
+        isVerified: true,
+      },
+      // organisation: Organisation | undefined;
+      profile: {
+        displayName: idp.rawInformation?.User?.name ?? "",
+        firstName: idp.rawInformation?.User?.given_name ?? "",
+        lastName: idp.rawInformation?.User?.family_name ?? "",
+      },
+      idpLinks: [idpLink],
+    };
+    return req;
+  },
+  [ProviderSlug.GITHUB]: (idp: IDPInformation) => {
+    const idpLink: IDPLink = {
+      idpId: idp.idpId,
+      userId: idp.userId,
+      userName: idp.userName,
+    };
+    const req: Partial<AddHumanUserRequest> = {
+      username: idp.userName,
+      email: {
+        email: idp.rawInformation?.email,
+        isVerified: true,
+      },
+      // organisation: Organisation | undefined;
+      profile: {
+        displayName: idp.rawInformation?.name ?? "",
+        firstName: idp.rawInformation?.name ?? "",
+        lastName: idp.rawInformation?.name ?? "",
+      },
+      idpLinks: [idpLink],
+    };
+    return req;
+  },
+};
+
+function retrieveIDP(
+  id: string,
+  token: string
+): Promise<IDPInformation | undefined> {
+  const userService = user.getUser(server);
+  return userService
+    .retrieveIdentityProviderInformation({ intentId: id, token: token }, {})
+    .then((resp: RetrieveIdentityProviderInformationResponse) => {
+      return resp.idpInformation;
+    });
+}
+
+function createUser(
+  provider: ProviderSlug,
+  info: IDPInformation
+): Promise<string> {
+  const userData = (PROVIDER_MAPPING as any)[provider](info);
+  const userService = user.getUser(server);
+  return userService.addHumanUser(userData, {}).then((resp) => resp.userId);
+}
+
+export default async function Page({
+  searchParams,
+  params,
+}: {
+  searchParams: Record<string | number | symbol, string | undefined>;
+  params: { provider: ProviderSlug };
+}) {
+  const { id, token } = searchParams;
+  const { provider } = params;
+
+  if (provider && id && token) {
+    return retrieveIDP(id, token)
+      .then((information) => {
+        if (information) {
+          return createUser(provider, information).catch((error) => {
```

Comment:
do we have to link users depending on the IDP config?

Reply:
yes.. the information is different on any provider

Reply:
So, don't we have to call the link endpoint somewhere?
https://zitadel.com/docs/apis/resources/user_service/user-service-add-idp-link#add-link-to-an-identity-provider-to-an-user

Reply:
idpLinks: [idpLink] does the linking

Reply:
ah ok, :+1: